lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -299,11 +299,90 @@ namespace Js
+        Var *const newInlineSlots = reinterpret_cast<Var *>(reinterpret_cast<uintptr_t>(object) + this->GetOffsetOfObjectHeaderInlineSlots());
+        PropertyIndex i = 0;
+        while (i < this->GetInlineSlotCapacity())
+        {
+            newInlineSlots[i] = oldInlineSlots[i];
+            i++;
+        }
+
+        // auxSlot should only have 2 entry, move that to inlineSlot
+        newInlineSlots[i] = auxSlotZero;
+        newInlineSlots[i + 1] = auxSlotOne;
+
+        Assert(i == this->GetInlineSlotCapacity());
+    }
+
+    BOOL PathTypeHandlerBase::DeleteLastProperty(DynamicObject *const object)
+    {
+        // We are deleting the last property from PathTypeHandler
+        DynamicType* predecessorType = this->GetPredecessorType();
+
+        // -----------------------------------------------------------------------------------------
+        //     Current Type  | Predecessor Type |                  Action
+        // -----------------------------------------------------------------------------------------
+        //          OHI      |      OHI         | No movement needed
+        // -----------------------------------------------------------------------------------------
+        //          OHI      |    Not OHI       | Not possible (Should be a BUG)
+        // -----------------------------------------------------------------------------------------
+        //        Not OHI    |    Not OHI       | No movement needed
+        // -----------------------------------------------------------------------------------------
+        //        Not OHI    |      OHI         | Move from auxSlots to inline slots (ReOptimize)
+        // -----------------------------------------------------------------------------------------
+
+        bool isCurrentTypeOHI = this->IsObjectHeaderInlinedTypeHandlerUnchecked();
+
+        Assert(predecessorType->GetTypeHandler()->IsPathTypeHandler());
+        PathTypeHandlerBase* predecessorTypeHandler = (PathTypeHandlerBase*)predecessorType->GetTypeHandler();
+
+        Assert(predecessorTypeHandler->GetUnusedBytesValue() == (this->GetUnusedBytesValue() - 1));
+
+        bool isPredecessorTypeOHI = predecessorTypeHandler->IsObjectHeaderInlinedTypeHandlerUnchecked();
+
+        AssertMsg(!isCurrentTypeOHI || isPredecessorTypeOHI, "Current type is ObjectHeaderInlined but previous type is not ObjectHeaderInlined?");
+
+        if (!isCurrentTypeOHI && isPredecessorTypeOHI)
+        {
+            Assert(predecessorTypeHandler->GetInlineSlotCapacity() == (this->GetUnusedBytesValue() - 1));
+            this->MoveAuxSlotsToObjectHeader(object);
+        }
+
+#if DBG
+        if ((isCurrentTypeOHI && isPredecessorTypeOHI) || (!isCurrentTypeOHI && !isPredecessorTypeOHI))
+        {
+            AssertMsg(this->GetInlineSlotCapacity() == predecessorTypeHandler->GetInlineSlotCapacity(), "InlineSlotCapacity of types should match");
+        }
```

Comment:
can you instead have assert `DynamicObject::IsTypeHandlerCompatibleForObjectHeaderInlining(oldtypeHandler, newTypeHandler)` . It validates much more than just inlineSlotCapacity.